### PR TITLE
Add breaking changes doc for v2.2.0

### PR DESCRIPTION
**Pull request by super-harsh**
**What this PR does / why we need it**:

Adds breaking changes documentation and steps for upgrading.

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+title: "v2.2.0 Breaking Changes"
+linkTitle: "v2.2.0"
+weight: 80
+---
+
+## ManagedClusters ManagedClusterServicePrincipalProfile.Secret field is now marked as a SecretReference
+  We always try to avoid breaking changes, but in this case, allowing raw passwords in the spec is a security problem and as such we've
+  decided to make a break to correct this issue.
+
+**Action required:** If the `ContainerService/ManagedClusters` resource is installed in your cluster and the `ManagedClusterServicePrincipalProfile.Secret` property is set on your `ManagedCluster` resource, below steps should be performed before upgrading:
```
Comment (theunrepentantgeek) [inline docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md:11]:
```suggestion
**Action required:** If the `ContainerService/ManagedClusters` resource is used in your cluster and the `ManagedClusterServicePrincipalProfile.Secret` property is set, do the following before upgrading ASO:
```

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+1. Annotate the resource with `serviceoperator.azure.com/reconcile-policy: skip` to prevent ASO from trying to reconcile the resource while you are upgrading.
+2. Download the current YAML for the resource using `kubectl` if you don't have it elsewhere.
+3. Create a kubernetes secret containing the value for `ManagedClusterServicePrincipalProfile.Secret`.
+4. Edit downloaded YAML in step 2, and add a secret key and name reference. Example [here](https://github.com/Azure/azure-service-operator/blob/main/v2/samples/compute/v1api/v1api20201201_virtualmachine.yaml#L18).
```
Comment (theunrepentantgeek) [inline docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md:16]: Can this be done before upgrading, or does this have to happen afterwards?

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+5. Delete the resource from your cluster using `kubectl delete`. Your Azure resource will be left untouched because of the `reconcile-policy` annotation you added above.
+6. [Upgrade ASO]( {{< relref "upgrading" >}} ) in your cluster.
+7. Apply the updated YAML to your cluster using `kubectl apply`. If any errors occur, address them.
+8. If the `reconcile-policy` annotation is still present, remove it from the resource.
+
+## Removed un-used Status properties
+
+The below fields are never returned from the service and end up being an empty string always. The changes here do not affect the users, hence **no action is required**.
```
Comment (theunrepentantgeek) [inline docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md:24]: Do these end up as empty strings or nulls?

```suggestion
These fields are never returned from the service and end up being an empty string always. The changes here do not affect the users, hence **no action is required**. 
```

**Review by theunrepentantgeek — COMMENTED** [review]
A couple of suggestions and a question. We should update the [index page](https://azure.github.io/azure-service-operator/guide/breaking-changes/) for breaking changes  in this PR as well.

**Review by theunrepentantgeek — APPROVED** [review]

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md (hunk 2 above)
Comment (super-harsh) [reply docs/hugo/content/guide/breaking-changes/breaking-changes-v2.2.0.md:16]: Can be done anytime as at this time, the user is going to update the local copy of their YAML

**Review by theunrepentantgeek — APPROVED** [review]

**Pull request merged**